drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -370,16 +402,89 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 	}
 
 	if (i == clk_values) {
-		dev_err(bus->dev, "%s: could not find clock value for bandwidth %d\n",
+multilane:
+		dev_dbg(bus->dev, "%s: could not find clock value for bandwidth %d, checking multi-lane\n",
 			__func__, bus->params.bandwidth);
-		return -EINVAL;
+
+		m_rt = list_last_entry(&bus->m_rt_list, struct sdw_master_runtime, bus_node);
+		s_rt = list_first_entry(&m_rt->slave_rt_list, struct sdw_slave_runtime, m_rt_node);
+		slave_prop = &s_rt->slave->prop;
+
+		/*
+		 * Find an available manager lane for the first Peripheral
+		 * Testing the first Peripheral is enough because we can't use multi-lane
+		 * if we can't find any available lane for the first Peripheral
```

Comment:
I don't follow the logic.

If we can't find any available lanes for the first Peripheral, then we can't use multi-lane.
Buf if we want one available lane for the first Peripheral, we still need to check if other peripherals can use this lane.

Edit: and later you actually test all lanes so the comment is misleading. You *start* by testing the first peripheral and later verify that that this configuration works for all peripherals.

Reply:
The logic here is to find an available lane from the first Peripheral, then check if the lane is available to all Peripherals.
The comment is to explain why I use `s_rt = list_first_entry(&m_rt->slave_rt_list, struct sdw_slave_runtime, m_rt_node);`